### PR TITLE
CAT-1347 erb to epp conversion

### PR DESCRIPTION
**Pull request by dinesh-gujar**
## Summary
Converting ERB files to EPP for performance upgrade.

### CODE REVIEW

File: templates/vhost/_http2.epp
Comment (ekohl) [inline templates/vhost/_http2.epp:7]: I think this style is a bit confusing. One uses `if` and the other `unless`, while both doing the same thing. I'd align on `if $param != undef` consistently throughout this file.

File: templates/vhost/_filters.epp
Comment (ekohl) [inline templates/vhost/_filters.epp:4]: Apache 2.2 support was dropped

```suggestion
  ## as per http://httpd.apache.org/docs/2.4/mod/mod_filter.html
```

File: templates/vhost/_http2.epp
Comment (dinesh-gujar) [reply templates/vhost/_http2.epp:7]: or unless in whole file ... i think that would be minimal changes ..thoughts ?

File: templates/vhost/_filters.epp
Comment (dinesh-gujar) [reply templates/vhost/_filters.epp:4]: Thanks for the update will resolve in upcoming commits ..

File: templates/vhost/_http2.epp
Comment (smortex) [reply templates/vhost/_http2.epp:7]: +1 for using the same syntax to "output this if $foo is not undef"

File: templates/vhost/_suexec.epp
Comment (smortex) [inline templates/vhost/_suexec.epp:4]: Missing `\n` at EOF

File: templates/vhost/_aliases.epp
Comment (dinesh-gujar) [reply templates/vhost/_aliases.epp:14]: sure will check that and update all the files accordingly

File: templates/vhost/_http2.epp
Comment (dinesh-gujar) [reply templates/vhost/_http2.epp]: @ekohl  done with changes so marking as resolved.

File: templates/vhost/_aliases.epp
Comment (dinesh-gujar) [reply templates/vhost/_aliases.epp]: Done

File: templates/vhost/_suexec.epp
Comment (dinesh-gujar) [reply templates/vhost/_suexec.epp]: Done

File: templates/vhost/_filters.epp
Comment (dinesh-gujar) [reply templates/vhost/_filters.epp]: Done

**Comment by Ramesh7** [comment]
can you please update your branch with main?

**Comment by dinesh-gujar** [comment]
> can you please update your branch with main?

yes just waiting for review for my changes once approved will do that and will also do the rebase

File: templates/vhost/_aliases.epp
Comment (Ramesh7) [inline templates/vhost/_aliases.epp:6]: maint - spacing between closing braces and `elsif`.

File: templates/vhost/_aliases.epp
Comment (Ramesh7) [inline templates/vhost/_aliases.epp:10]: maint - 
```
<%- } elsif $alias_statement["scriptaliasmatch"] {-%>
```

File: templates/vhost/_custom_fragment.epp
Comment (Ramesh7) [inline templates/vhost/_custom_fragment.epp:1]: maint - 
```
<% if $custom_fragment { -%>
```

File: templates/vhost/_docroot.epp
Comment (Ramesh7) [inline templates/vhost/_docroot.epp:3]: maint - 
```
<% if $virtual_docroot { -%>
```

File: templates/vhost/_docroot.epp
Comment (Ramesh7) [inline templates/vhost/_docroot.epp:6]: maint - 
```
<% if $docroot and ((! $virtual_docroot) or $virtual_use_default_docroot) { -%>
```

File: templates/vhost/_fallbackresource.epp
Comment (Ramesh7) [inline templates/vhost/_fallbackresource.epp:1]: maint - 
```
<% if $fallbackresource { -%>
```

File: templates/vhost/_file_footer.epp
Comment (Ramesh7) [inline templates/vhost/_file_footer.epp:1]: maint - 
```
<% $define.each |$k, $v| { -%>
```

File: templates/vhost/_file_footer.epp
Comment (Ramesh7) [inline templates/vhost/_file_footer.epp:6]: The original variable name is `$passenger_pre_start` and we are also creating another local variable in scope with same name, which increases the readability. Can we name it something diff. Eg : 
```
<% if $passenger_pre_start { -%>
  <%- [$passenger_pre_start].flatten().filter |$value| { $value }.each |$value| { -%>
PassengerPreStart <%= $value %>
  <%- } -%>
<% } -%>
```

File: templates/vhost/_file_header.epp
Comment (Ramesh7) [inline templates/vhost/_file_header.epp:2]: can we follow right spacing for opening and closing if else blocks?
I don't see it consistent here?

File: templates/vhost/_filters.epp
Comment (Ramesh7) [inline templates/vhost/_filters.epp:1]: maint - 
```
<% unless $filters.empty { -%>

  ## Filter module rules
  ## as per http://httpd.apache.org/docs/2.4/mod/mod_filter.html
  <%- $filters.each |$filter| { -%>
  <%= $filter %>
  <%- } -%>
<% } -%>
```

File: templates/vhost/_docroot.epp
Comment (dinesh-gujar) [reply templates/vhost/_docroot.epp]: done

File: templates/vhost/_docroot.epp
Comment (dinesh-gujar) [reply templates/vhost/_docroot.epp]: done

File: templates/vhost/_file_footer.epp
Comment (dinesh-gujar) [reply templates/vhost/_file_footer.epp]: done

File: templates/vhost/_header.epp
Comment (dinesh-gujar) [reply templates/vhost/_header.epp]: done

File: CHANGELOG.md
Comment (ekohl) [inline CHANGELOG.md:667]: This is an odd thing to change. I'd keep the changelog as it is, so it matches the actual PR title.

File: templates/vhost/_ssl.erb
Comment (ekohl) [inline templates/vhost/_ssl.erb:9]: Please keep the indenting consistent throughout this file. It's way more important that the final rendered file looks consistent than that the if/else indenting matches.
```suggestion
  SSLCertificateChainFile "<%= @ssl_chain %>"
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: templates/vhost/_ssl.erb
Comment (dinesh-gujar) [reply templates/vhost/_ssl.erb:10]: sure

File: CHANGELOG.md
Comment (dinesh-gujar) [reply CHANGELOG.md:667]: sure

File: templates/vhost/_filters.epp
Comment (dinesh-gujar) [reply templates/vhost/_filters.epp]: done

File: templates/vhost/_ssl.erb
Comment (dinesh-gujar) [reply templates/vhost/_ssl.erb]: done

File: templates/vhost/_charsets.epp
Comment (ekohl) [inline templates/vhost/_charsets.epp:1]:
```suggestion
<% if $add_default_charset { -%>
```

File: templates/vhost/_filters.epp
Comment (ekohl) [inline templates/vhost/_filters.epp:5]:
```suggestion
  <%- $filters.each |$filter| { -%>
```

File: templates/vhost/_header.epp
Comment (ekohl) [inline templates/vhost/_header.epp:5]:
```suggestion
  <%- $headers.each |$header_statement| { -%>
```

File: templates/vhost/_itk.epp
Comment (ekohl) [inline templates/vhost/_itk.epp:20]: I don't think this works and needs to be
```suggestion
     <%- if $facts['kernelversion'] >= '3.5.0' { -%>
```

File: templates/vhost/_wsgi.epp
Comment (ekohl) [inline templates/vhost/_wsgi.epp:8]: AFAIK you don't need to use `type()` and `Type`.
```suggestion
<% if $wsgi_daemon_process =~ String and $wsgi_daemon_process_options {-%>
```

**Review by ekohl — CHANGES_REQUESTED** [review]
You use `{-%>` all the time, but (like @Ramesh7) I think it should be `{ -%>`. I marked a few, but it's almost in every file.

File: templates/vhost/_wsgi.epp
Comment (dinesh-gujar) [reply templates/vhost/_wsgi.epp]: done

File: templates/vhost/_itk.epp
Comment (dinesh-gujar) [reply templates/vhost/_itk.epp]: done

File: templates/vhost/_header.epp
Comment (dinesh-gujar) [reply templates/vhost/_header.epp]: Done

File: templates/vhost/_filters.epp
Comment (dinesh-gujar) [reply templates/vhost/_filters.epp]: Done

File: templates/vhost/_charsets.epp
Comment (dinesh-gujar) [reply templates/vhost/_charsets.epp]: Done

File: templates/vhost/_file_header.epp
Comment (ekohl) [inline templates/vhost/_file_header.epp:8]: I think `type()` and `Type` aren't needed
```suggestion
  <%- if $mdomain =~ String { -%>
```

File: templates/vhost/_scriptalias.epp
Comment (ekohl) [inline templates/vhost/_scriptalias.epp:7]: I'd consider this for readability. Note that line 1 is then redundant.
```suggestion
<%- $aliases1 = if $scriptalias { [{ 'alias' => '/cgi-bin', 'path' => $scriptalias }] } else { [] } -%>
<%- ($scriptaliases + $aliases1).each |$salias| { -%>
```

**Review by ekohl — COMMENTED** [review]
Minor notes. I'd also like to see data types in the EPP templates, but wouldn't block on that.

File: templates/vhost/_scriptalias.epp
Comment (dinesh-gujar) [reply templates/vhost/_scriptalias.epp:7]: i think they are updating existing array of hash on condition this new hash will get added to that array ..else it will have old values

File: manifests/vhost.pp
Comment (dinesh-gujar) [reply manifests/vhost.pp:2244]: yeah will remove that as well

File: templates/vhost/_file_header.epp
Comment (dinesh-gujar) [reply templates/vhost/_file_header.epp]: Done

File: templates/vhost/_scriptalias.epp
Comment (dinesh-gujar) [reply templates/vhost/_scriptalias.epp:7]: Done

File: manifests/vhost.pp
Comment (dinesh-gujar) [reply manifests/vhost.pp]: Done

File: manifests/vhost.pp
Comment (ekohl) [inline manifests/vhost.pp:2401]: This template hasn't been converted. Is that an omission? For ERB templates I'd keep the "Template uses" comment in.

Comment (dinesh-gujar) [reply manifests/vhost.pp:2401]: yes that is an omission

File: templates/vhost/_file_header.epp
Comment (ekohl) [inline templates/vhost/_file_header.epp:43]: Please use a line ending at the end of the file.

Comment (dinesh-gujar) [reply templates/vhost/_file_header.epp]: Done

File: manifests/vhost.pp
Comment (ekohl) [inline manifests/vhost.pp:2370]: This still uses an ERB template.

File: manifests/vhost.pp
Comment (ekohl) [inline manifests/vhost.pp:2438]: Did you intend to change this too?

File: manifests/vhost.pp
Comment (ekohl) [inline manifests/vhost.pp:2404]: This is just a single line. I'd inline it:
```suggestion
    content => "  ServerSignature Off\n",
```

**Review by ekohl — DISMISSED** [review]

File: templates/vhost/_file_header.epp
Comment (dinesh-gujar) [reply templates/vhost/_file_header.epp]: @ekohl

File: manifests/vhost.pp
Comment (dinesh-gujar) [reply manifests/vhost.pp:2404]: i tried it was failing the acceptance ... feel free to give another shot from your side.

Comment (ekohl) [reply manifests/vhost.pp:2404]: https://github.com/puppetlabs/puppetlabs-apache/pull/2467

File: manifests/vhost.pp
Comment (dinesh-gujar) [reply manifests/vhost.pp:2370]: Reverted

File: manifests/vhost.pp
Comment (dinesh-gujar) [reply manifests/vhost.pp]: Reverted

Comment (dinesh-gujar) [reply manifests/vhost.pp]: Done @ekohl

File: manifests/vhost.pp
Comment (dinesh-gujar) [reply manifests/vhost.pp]: @ekohl  I checked this pipeline and saw that acceptance are passing so accommodating the same changes in this MR as well

**Comment by dinesh-gujar** [comment]
@ekohl  resolved the comments.

**Comment by dinesh-gujar** [comment]
@ekohl  please review

File: templates/vhost/_scriptalias.epp
Comment (ekohl) [inline templates/vhost/_scriptalias.epp:1]: Note that scriptaliases is deprecated anyway. There's going to be a major release, so I'd first clean that up. https://github.com/puppetlabs/puppetlabs-apache/pull/2469 does that.

Comment (dinesh-gujar) [reply templates/vhost/_scriptalias.epp]: @ekohl  resolved!  please review

**Review by praj1001 — APPROVED** [review]

**Comment by praj1001** [comment]
LGTM

**Review by praj1001 — APPROVED** [review]

**Comment by dinesh-gujar** [comment]
1 requested change is done but still showing up in git hub so dismissing review

**Pull request merged**
